lib/qless/worker/base.rb

```diff
@@ -15,13 +15,16 @@ module Workers
 
     class BaseWorker
       attr_accessor :output, :reserver, :log_level, :interval, :paused,
-                    :options
+                    :options, :sighup_handler
 
       def initialize(reserver, options = {})
         # Our job reserver and options
         @reserver = reserver
         @options = options
 
+        # SIGHUP handler
+        @sighup_handler = options.fetch(:sighup_handler) { lambda { } }
```

Comment:
It still stands. SIGHUP is actually work of engineer, not the gem.  And the goal is to investigate what the hell worker is doing (which rabbithole is it in).